starport-core/src/main/scala/com/krux/starport/config/StarportSettings.scala

```diff
@@ -20,8 +18,19 @@ class StarportSettings(val config: Config) extends Serializable {
 
   val starportNotificationSns = config.getString("krux.starport.notification.sns")
 
-  val metricSettings: Option[Config] =
-    Try(config.getConfig("krux.starport.metric.graphite")).toOption
+  val metricsEngine: MetricSettings =
+    Try(config.getConfig("krux.starport.metric.engine")) match {
+      case Success(engine) => engine match {
+        case s if (s == "graphite") => GraphiteReporterSettings
+        case s if (s == "cloudwatch") => CloudWatchReporterSettings
+      }
+      case Failure(_) => throw new InstantiationException("No metrics engine specified.")
+    }
+
+  val metricSettings: Option[Config] = metricsEngine match {
```

Comment:
I found the name a bit misleading because this is not a `MetricSettings`. I'd rename this to something like `metricConfig`.

Reply:
That's a much better name.